src/kOS.Safe/Compilation/KS/KSScript.cs

```diff
@@ -26,8 +26,16 @@ public override List<CodePart> Compile(string filePath, int startLineNum, string
                 var compiler = new Compiler();
                 LoadContext(contextId);
 
-                // TODO: handle compile errors (e.g. wrong run parameter count)
-                CodePart mainPart = compiler.Compile(startLineNum, parseTree, currentContext, options);
+                CodePart mainPart;
+                try
+                {
+                    mainPart = compiler.Compile(startLineNum, parseTree, currentContext, options);
+                }
+                catch (KOSCompileException e)
+                {
+                    e.AddSourceText((short)startLineNum, scriptText);
+                    throw e;
```

Comment:
i assume you are trying to throw the same exception with the original origin? if so you should remove the `e`

```catch (KOSCompileException e)
                {
                    e.AddSourceText((short)startLineNum, scriptText);
                    throw;
}```